src/rdkafka_fetcher.c

```diff
@@ -673,7 +816,18 @@ rd_kafka_fetch_reply_handle(rd_kafka_broker_t *rkb,
         }
 
         /* Top level tags */
-        rd_kafka_buf_skip_tags(rkbuf);
+        if (leader_tags_present) {
+                rd_kafka_buf_read_uvarint(rkbuf, &_tagcnt);
+                for (i = 0; i < (int)_tagcnt; i++) {
+                        rd_kafka_buf_read_uvarint(rkbuf, &_tag);
+                        rd_kafka_buf_read_uvarint(rkbuf, &_taglen);
+                        if (rd_kafka_fetch_reply_handle_read_tag(
+                                rkbuf, _tag, _taglen, &fetch_tags) == -1)
+                                goto err_parse;
+                }
+                rd_kafka_handle_Fetch_metadata_update(rkb, &fetch_tags);
+        } else if (rd_kafka_buf_ApiVersion(request) >= 12)
+                rd_kafka_buf_skip_tags(rkbuf);
```

Comment:
This becomes just
```suggestion
        rd_kafka_buf_read_tags(rkbuf, rd_kafkap_Fetch_reply_tags_parse,
                               &FetchTags);
```

Reply:
Before returning in the success case add
```c
        rd_kafka_handle_Fetch_metadata_update(rkb, &FetchTags);

        rd_kafkap_Fetch_reply_tags_destroy(&FetchTags);
```

Reply:
In the error case also destroy the FetchTags, before `BADMSG`

Reply:
Updated